test/boost/bptree_test.cc

```diff
@@ -356,3 +358,123 @@ BOOST_AUTO_TEST_CASE(test_avx_search) {
     t.clear();
     check_conversions();
 }
+
+using test_key = tree_test_key_base;
+
+class test_data {
+    int _value;
+public:
+    test_data() : _value(0) {}
+    test_data(test_key& k) : _value((int)k + 10) {}
+
+    operator unsigned long() const { return _value; }
+    bool match_key(const test_key& k) const { return _value == (int)k + 10; }
+};
+
+template <> struct fmt::formatter<test_data> : fmt::formatter<string_view> {
+    auto format(test_data d, fmt::format_context& ctx) const {
+        return fmt::format_to(ctx.out(), "{}", static_cast<unsigned long>(d));
+    }
+};
+
+BOOST_AUTO_TEST_CASE(stress_test) {
```

Comment:
> what if we have performance tests for the b+ tree, are we going to put them into this file?

We already have it, check `test/perf/perf_collection.cc`

> i don't object to this change. just don't feel right if we commingle them.

Well, yes, one of the reasons why these tests were in unit/ suite was the wish to parametrise them somehow. Now when they are in boost/ those parameters are set to ex-defaults with the only way to configure by editing and re-compiling

Reply:
Please note that we also have the manual/ subdirectory, for tests that are NOT part of the test.py CI and need to be run manually - with parameters saying what to test. Maybe some of these tests should move there. But this should be done after careful thought. It is also possible that some of these tests had parameters but nobody ever used them and everyone just assumed that they run in CI with the default parameters. If that's the case - then indeed moving them to boost/ makes sense.

Reply:
> It is also possible that some of these tests had parameters but nobody ever used them and everyone just assumed that they run in CI with the default parameters.

It's pretty much the case here. When I developer the collections, I wrote those stress tests and ran them with different parameters. Then put the "larges scale" ones as defaults. Since then no changes in the collections code was made, respectively, only CI is running those tests.

Reply:
> Why not combine both approaches? 

Because boost/ and manual/ are two different suites, we can't have one test in both without ... some effort on sharing the code ... oh, well. 

I'd prefer a way to pass parameters to a boost/ test , @nyh [says](https://github.com/scylladb/scylladb/pull/20475#issuecomment-2342966578) it's somehow possible

Reply:
> > Why not combine both approaches?
> 
> Because boost/ and manual/ are two different suites, we can't have one test in both without ... some effort on sharing the code ... oh, well.
> 
> I'd prefer a way to pass parameters to a boost/ test , @nyh [says](https://github.com/scylladb/scylladb/pull/20475#issuecomment-2342966578) it's somehow possible

Well, that's understandable that it will not work out of the box, and some additional work needed. I am not sure how much work needed, but it doesn't seem too much.
I have no objection with @nyh's way handling this. Furthermore, I suppose it's even better way than I proposed.